lib/web_ui/test/engine/pointer_binding_test.dart

```diff
@@ -114,9 +129,272 @@ void main() {
+      expect(packets[5].data[0].pointerIdentifier, equals(1));
+      expect(packets[5].data[0].synthesized, equals(false));
+      expect(packets[5].data[0].physicalX, equals(20.0));
+      expect(packets[5].data[0].physicalY, equals(10.0));
+      expect(packets[5].data[0].physicalDeltaX, equals(-20.0));
+      expect(packets[5].data[0].physicalDeltaY, equals(-20.0));
+
+      expect(packets[6].data, hasLength(1));
+      expect(packets[6].data[0].change, equals(ui.PointerChange.down));
+      expect(packets[6].data[0].pointerIdentifier, equals(2));
+      expect(packets[6].data[0].synthesized, equals(false));
+      expect(packets[6].data[0].physicalX, equals(20.0));
+      expect(packets[6].data[0].physicalY, equals(10.0));
+      expect(packets[6].data[0].physicalDeltaX, equals(0.0));
+      expect(packets[6].data[0].physicalDeltaY, equals(0.0));
+    });
+
+    test('does synthesize add or hover or more for scroll', () {
+      List<ui.PointerDataPacket> packets = <ui.PointerDataPacket>[];
+      ui.window.onPointerDataPacket = (ui.PointerDataPacket packet) {
+        packets.add(packet);
+      };
+
+      glassPane.dispatchEvent(html.WheelEvent('wheel',
+        button: 1,
+        clientX: 10,
+        clientY: 10,
+        deltaX: 10,
+        deltaY: 10,
+      ));
+
+      glassPane.dispatchEvent(html.WheelEvent('wheel',
+        button: 1,
+        clientX: 20,
+        clientY: 50,
+        deltaX: 10,
+        deltaY: 10,
+      ));
+
+      glassPane.dispatchEvent(html.PointerEvent('pointerdown', {
+        'pointerId': -1,
+        'button': 1,
+        'clientX': 20.0,
+        'clientY': 50.0,
+      }));
+
+      glassPane.dispatchEvent(html.WheelEvent('wheel',
+        button: 1,
+        clientX: 30,
+        clientY: 60,
+        deltaX: 10,
+        deltaY: 10,
+      ));
+
+      expect(packets, hasLength(7));
+
+      // An add will be synthesized.
+      expect(packets[0].data, hasLength(2));
+      expect(packets[0].data[0].change, equals(ui.PointerChange.add));
+      expect(packets[0].data[0].pointerIdentifier, equals(0));
+      expect(packets[0].data[0].synthesized, equals(true));
+      expect(packets[0].data[0].physicalX, equals(10.0));
+      expect(packets[0].data[0].physicalY, equals(10.0));
+      expect(packets[0].data[0].physicalDeltaX, equals(0.0));
+      expect(packets[0].data[0].physicalDeltaY, equals(0.0));
+
+      expect(packets[0].data[1].change, equals(ui.PointerChange.hover));
+      expect(packets[0].data[1].signalKind, equals(ui.PointerSignalKind.scroll));
+      expect(packets[0].data[1].pointerIdentifier, equals(0));
+      expect(packets[0].data[1].synthesized, equals(false));
+      expect(packets[0].data[1].physicalX, equals(10.0));
+      expect(packets[0].data[1].physicalY, equals(10.0));
+      expect(packets[0].data[1].physicalDeltaX, equals(0.0));
+      expect(packets[0].data[1].physicalDeltaY, equals(0.0));
+      // Scroll event will fire twice
```

Comment:
I am seeing this behavior which i found it weird. glassPane.dispatchEvent(html.WheelEvent(..)) will actually receive two scroll event back to back in glassPaneElement listener

Reply:
/cc @mdebbar 

Hmm, could we possibly have the pointer binding double-subscribed on the glasspane?

Reply:
It seems like the some tests do not clear the subscription in between test. It will fire more event if i ran the entire test suite

Reply:
Ah yes, if it's over-firing in tests, then some tests are probably not cleaning up after themselves.